konfera/tests/test_models.py

```diff
@@ -193,27 +214,38 @@ def test_string_representation(self):
         )
 
     def test_automatic_order_generator(self):
-        time = timezone.now()
-        location = models.Location(title='test_title', street='test_street', city='test_city', postcode=000000,
-                                   state='test_state', capacity=20)
-        location.save()
-        event = models.Event(title='test_event', description='test', event_type='meetup',
-                             status=models.event.PUBLISHED, location=location, date_from=time, date_to=time)
-        event.save()
-        ticket_type = models.TicketType(title='test', description='test', price=100, event=event,
-                                        date_from=timezone.now(), date_to=timezone.now())
-        ticket_type.save()
-        discount_code = models.DiscountCode(title='test discount', hash='test', discount=60,
-                                            date_from=time, date_to=time, usage=1,
-                                            ticket_type=ticket_type)
-        discount_code.save()
-        ticket = models.Ticket(status='requested', title='mr', first_name="test", last_name="Test", type=ticket_type,
-                               email='test@test.com', phone='0912345678', discount_code=discount_code)
+
+        ticket = models.Ticket(status='requested', title='mr', first_name="test", last_name="Test",
+                               type=self.ticket_type, email='test@test.com', phone='0912345678',
+                               discount_code=self.discount_code)
         ticket.save()
         self.assertEquals(ticket.order.status, 'awaiting_payment')
-        self.assertEquals(ticket.order.price, ticket_type.price)
+        self.assertEquals(ticket.order.price, self.ticket_type.price)
         self.assertEquals(ticket.order.discount, 60)
 
+    def test_clean_and_save_ticket_discount_code(self):
+
+        # if cleaned and saved with no discount_code, they should both be successful
+        ticket_no_code = models.Ticket(status='requested', title='mr', first_name="test", last_name="Test",
+                                       type=self.ticket_type, email='test@test.com', phone='0912345678')
+        ticket_no_code.clean()
```

Comment:
I took out the calls to "clean" throughout the test. You're right, since clean is being called from save, we can just test it indirectly by calling save.